### PR TITLE
nixos/security/ca: enable support for compatibility bundles

### LINKED ISSUE #286722: Allow cacert to output something else than OpenSSL-specific PEM variant
### Describe the bug
Currently `pkgs.cacert` outputs most CA certificates using Openssl specific PEM format :

```
ACCVRAIZ1
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----

Trusted for:
  - 1.3.6.1.5.5.7.3.1 (RFC5280: serverAuth key usage)
  - 1.3.6.1.5.5.7.3.4 (RFC5280: emailProtection key usage)
-----BEGIN TRUSTED CERTIFICATE-----
...
-----END TRUSTED CERTIFICATE-----

AC RAIZ FNMT-RCM
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----

Trusted for:
  - 1.3.6.1.5.5.7.3.1 (RFC5280: serverAuth key usage)
-----BEGIN TRUSTED CERTIFICATE-----
...
...

```

Somehow, `BEGIN TRUSTED CERTIFICATE` format is not well supported by many applications and systems find themselves unable to load the nix CA bundle.

I currently have to split all these certificates from bundle file in post-processing, then apply the following openssl command to rebuild a new CA bundle containing only `BEGIN CERTIFICATE`  certs:

```bash
    ...
    # convert specific openssl x509 to real pem format
    for i in $(ls splitted-cacert-*); do
      openssl x509 -in $i -outform PEM -out $i
    done
    
    # rebuild the final ca-bundle
    cat splitted-cacert-* > $out/ca-bundle.crt
```

### Expected behavior
If needed, I'd like to have the option to ask upstream `cacert` package to **avoid using OpenSSL-specific format and generate a "normal" pem formatted bundleCA (without any additionnal hussle)**

### Notify maintainers
@RaitoBezarius @fpletz

Thanks a lot !

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

Certain software stacks have no support for OpenSSL non-standard PEM format and will fail to use our NixOS CA bundle.

For this, it is necessary to fallback on a 'compatibility' bundle, which will contain no additional trust rules.

Open questions:

- Add a NixOS test?
- Make it configurable at the `cacert` package level to save on space?
- Document it in the release notes as a highlight?

~~Depends on https://github.com/lukegb/buildcatrust/pull/4.~~ ~~Depends on stable release of buildcatrust.~~
Closes #286722.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
I will retarget staging.

File: nixos/modules/security/ca.nix [hunk 1]
```diff
@@ -23,6 +24,17 @@ in
       internal = true;
     };
 
+    security.pki.enableCompatibleBundle = mkEnableOption ''usage of a compatibility bundle.
```
Comment (lukegb) [inline nixos/modules/security/ca.nix:27]: Maybe `useCompatibilityBundle` instead?

**Review by lukegb — APPROVED** [review]

**Comment by lukegb** [comment]
You missed a reference to `enableCompatibleBundle` :)

**Comment by RaitoBezarius** [comment]
> You missed a reference to `enableCompatibleBundle` :)

"I had one job"

**Pull request merged**